### ISSUE TITLE
Keywords fail - "No keyword with name 'New Browser' found." when the Browser lib refers to a customjs 

### ISSUE THREAD

**Issue by ryanrosello-og** [damies13/rfswarm#129]
This test works (does not refer to custom js):

```
*** Settings ***

Library     Browser   

*** Test Cases ***
Starting a browser with a page
    New Browser    chromium    headless=true
    New Context    viewport={'width': 1920, 'height': 1080}
    New Page       https://marketsquare.github.io/robotframework-browser/Browser.html
    Get Title      ==    Browser
```

This **does not** (refers to a custom js):

```
*** Settings ***

Library     Browser   jsextension=${CURDIR}/../../resources/keywords/_custom_keywords/module.js

*** Test Cases ***
Starting a browser with a page
    New Browser    chromium    headless=true
    New Context    viewport={'width': 1920, 'height': 1080}
    New Page       https://marketsquare.github.io/robotframework-browser/Browser.html
    Get Title      ==    Browser
```

This test works when executed via robot:

![2023-02-14_12-46-52](https://user-images.githubusercontent.com/50574915/218627970-b367154e-1e1e-48ac-ba93-3387a733411d.jpg)

When I look at the csv output, the keywords are failing with:

`No keyword with name 'New Browser' found.`

**Environment:**

Windows 11
python  3.10.9
robotframework          6.0.2
robotframework-browser  15.2.0

**Results:**
[20230214_124215_scr2_simples_summary.csv](https://github.com/damies13/rfswarm/files/10728309/20230214_124215_scr2_simples_summary.csv)
[20230214_124215_scr2_simples_agent_data.csv](https://github.com/damies13/rfswarm/files/10728310/20230214_124215_scr2_simples_agent_data.csv)
[20230214_124215_scr2_simples_raw_result_data.csv](https://github.com/damies13/rfswarm/files/10728311/20230214_124215_scr2_simples_raw_result_data.csv)

**Comment by ryanrosello-og** [comment]
@damies13 Had a bit more of a look at this, it seems that the `.js` files are not getting copied across to the temp folder?
![2023-02-14_16-32-22](https://user-images.githubusercontent.com/50574915/218657823-ac7bb3bc-68a6-4418-8512-192f7a4744f4.jpg)

**Comment by damies13** [comment]
> it seems that the `.js` files are not getting copied across to the temp folder?

This is correct, the way your test case was setup the .js files wouldn't be copied.

Start by having a read of [Robot File handling (transfer from Manager to Agent)](https://github.com/damies13/rfswarm/blob/master/Doc/Overview.md#robot-file-handling-transfer-from-manager-to-agent) to understand the why.

Next refer to [Support Files](https://github.com/damies13/rfswarm/blob/master/Doc/Preparing_for_perf.md#support-files) 

You'll probably want to do something like so that the js file gets copied:
```
Metadata 	File 	${CURDIR}/../../resources/keywords/_custom_keywords/module.js
```

If module.js uses other .js files, then you might want to do something like
```
Metadata 	File 	${CURDIR}/../../resources/keywords/_custom_keywords/*.js
```

I'd never anticipated that a .js file would be needed, but I had allowed for any file type to be transferred to the agent, because you never know what people might need.

Dave.

**Comment by ryanrosello-og** [comment]
Unfortunately this still seems to be problematic for me.  I've created a dummy repo here

https://github.com/ryanrosello-og/rfswarm-repo

**Comment by damies13** [comment]
Hi Ryan,

Thanks for the dummy repo you provided it was helpful in identifying the problem

The issue is actually in the manager not the agent as I first suspected


![Screenshot from 2023-02-26 14-33-02](https://user-images.githubusercontent.com/15232871/221392452-2890bc16-2b7a-4977-86ef-14f11123329e.png)

`${CURDIR}/../../resources/keywords/_custom_keywords/module.js`
got resolved to 
`/home/dave/rfswarm-repo/tests/resources/keywords/_custom_keywords/module.js`
when it should have been resolved as 
`/home/dave/rfswarm-repo/resources/keywords/_custom_keywords/module.js`

As the file `/home/dave/rfswarm-repo/tests/resources/keywords/_custom_keywords/module.js` doesn't exist it never got hashed and so the agent didn't know to download it from the manager

As a workaround for this bug configure simple.robot like this:
```
*** Settings ***
Metadata    File    ../../resources/keywords/_custom_keywords/module.js
Library     Browser    jsextension=${CURDIR}/../../resources/keywords/_custom_keywords/module.js
*** Test Cases ***
Hello
    New Page
    Go To    https://playwright.dev
    ${result}=    myGoToKeyword    testing
    Should Be Equal As Strings    ${result}    this is the result
```

Configured like that I managed to get the test to run successfully

![Screenshot from 2023-02-26 14-44-32](https://user-images.githubusercontent.com/15232871/221392766-a64a7dde-011d-44f9-8849-c7fd45fbefca.png)

I've marked this as a bug as the Manager need to handle `${CURDIR}` better.

**Comment by damies13** [comment]
As best I can tell this is fixed and working on Linux, now need to spin up a windows VM and double check it

**Comment by damies13** [comment]
Now working on Windows as well

**Comment by damies13** [comment]
Merged into v1.0.1 release branch

**Comment by ryanrosello-og** [comment]
great, thanks for looking into this.
